bin/crew

```diff
@@ -486,20 +486,13 @@ def files (pkgName)
 end
 
 def whatprovides (regexPat)
-  fileArray = []
-  needle = regexPat.gsub(/-/,',').gsub(/,/,'\-')
-  Dir[CREW_META_PATH + '*.filelist'].each do |packageList|
-    packageName = File.basename packageList, '.filelist'
-    File.readlines(packageList).each do |line|
-      found = line[/#{needle}/] if line.ascii_only?
-      if found
-        fileLine = packageName + ': ' + line
-        unless fileArray.include? fileLine
-          fileArray.push(fileLine)
-        end
-      end
-    end
+  # Use grep version command to ascertain whether we have a working grep.
+  unless system('grep -V > /dev/null 2>&1')
+    abort 'Grep is not working. Please install it with \'crew install grep\''.lightred
   end
```

Comment:
This shouldn't be needed since grep is already installed out-of-the-box.

Reply:
It isn't in core though. Should we assume that system apps will always work? That caused a problem with curl.  (Luckily grep has fewer dependencies.)

Up to you. This is a failsafe warning which doesn't slow things down.